subprojects/build-cache/src/main/java/org/gradle/caching/internal/controller/DefaultBuildCacheController.java

```diff
@@ -93,89 +117,103 @@ public boolean isEmitDebugLogging() {
     }
 
     @Override
-    public <T> Optional<T> load(BuildCacheLoadCommand<T> command) {
-        final Unpack<T> unpack = new Unpack<>(command);
-
+    public Optional<BuildCacheLoadMetadata> load(BuildCacheKey key, CacheableEntity entity) {
+        Optional<BuildCacheLoadMetadata> result = Optional.empty();
         if (local.canLoad()) {
```

Comment:
We can now remove `LocalBuildCacheServiceHandle.canLoad()` as `load()` can simply return `Optional.empty()` if we can't.

Reply:
done